File: 1047.删除相邻重复项/Solution.java

public class Solution {

    public String removeDuplicates(String S) {
        int now = S.length();
        int next = 1;
        while (now != next) {
            now = S.length();
            S = S.replace("aa", "").replace("bb", "").replace("cc", "").replace("dd", "").replace("ee", "")
                    .replace("ff", "").replace("gg", "").replace("hh", "").replace("ii", "").replace("jj", "")
                    .replace("kk", "").replace("ll", "").replace("mm", "").replace("nn", "").replace("oo", "")
                    .replace("pp", "").replace("qq", "").replace("rr", "").replace("ss", "").replace("tt", "")
                    .replace("uu", "").replace("vv", "").replace("ww", "").replace("xx", "").replace("yy", "")
                    .replace("zz", "");
            next = S.length();
        }
        return S;
    }

    public static void main(String[] args) {

    }
}

File: 1047.删除相邻重复项/Solution.cpp
#include <iostream>
#include <stack>

using  namespace std;

class Solution
{
private:
public:
    string removeDuplicates(string S) {
        string ret="";
        for (auto c : S )
        {
            if (ret.back() == c )
            {
                ret+=c; 
            }
            else
            {
                 ret.pop_back();
            }
        }
        return ret;
    }

    //原字符串去重
    string removeDuplicates2(string S) {
        int top = 0;
        for (char ch : S) {
            if (top == 0 || S[top - 1] != ch) {
                S[top++] = ch;
            } else {
                top--;
            }
        }
        S.resize(top);
        return S;
    }
    
};
    int main(int argc, char const *argv[])
    {
        Solution s;
        cout<<s.removeDuplicates("abbaca");
        return 0;
    }

